src/tools/schema2kotlin.ts

```diff
@@ -181,6 +190,47 @@ class KotlinGenerator implements ClassGenerator {
     this.fieldSerializes.push(`"${field}" to ${fixed}.toReferencable()`);
     this.fieldDeserializes.push(`${fixed} = data.singletons["${fixed}"].toPrimitiveValue(${type}::class, ${defaultVal})`);
     this.fieldsForToString.push(`${fixed} = $${fixed}`);
+    if (isCollection) {
+      this.collectionSchemaFields.push(`"${field}" to ${typeMap[typeChar].schemaType}`);
+    } else {
+      this.singletonSchemaFields.push(`"${field}" to ${typeMap[typeChar].schemaType}`);
+    }
+
+  }
+
+  mapOf(items: string[]): string {
+    switch (items.length) {
+      case 0:
+        return `emptyMap()`;
+      case 1:
+        return `mapOf(${items[0]})`;
+      default:
+        return `\
+mapOf(
+${this.leftPad(items.join(',\n'), 4)}
```

Comment:
Could be worth making the indent amount an argument to this function.

Or rather, supply a `baseIndent` and add 4 to it for all the nested lines

Reply:
I like the idea of using a constant base indent everywhere. On a similar note, I'm going to break out this method into a Kotlin Codegen utilities file / class in a future PR.

Reply:
I might do it in this one, now that I think of it...

Reply:
Going to do this in a PR later today.